src/main/java/com/github/jcustenborder/kafka/connect/snmp/SnmpTrapSourceTask.java

```diff
@@ -91,35 +93,50 @@ public void start(Map<String, String> settings) {
 
     this.transport = setupTransport(this.config.listenAddress, this.config.listenProtocol, this.config.listenPort);
 
-    log.info("start() - Configuring ThreadPool DispatchPool to {} thread(s)", this.config.dispatcherThreadPoolSize);
-    this.threadPool = ThreadPool.create("DispatchPool", this.config.dispatcherThreadPoolSize);
-    this.messageDispatcher = createMessageDispatcher(this.threadPool, this.config.mpv3Enabled);
-    SecurityProtocols securityProtocols = setupSecurityProtocols(this.config.mpv3Enabled);
-
-    try {
-      this.transport.listen();
-    } catch (IOException e) {
-      throw new ConnectException("Exception thrown while calling transport.listen()", e);
+    if (this.config.snmp4jUseMultithreaded) {
+      log.info("start() - Configuring ThreadPool DispatchPool to {} thread(s)", this.config.dispatcherThreadPoolSize);
+      ThreadPool pool = ThreadPool.create("DispatchPool", this.config.dispatcherThreadPoolSize);
+      log.info("start() - Configuring multithreaded message dispatcher");
+      this.messageDispatcher = createMultiMessageDispatcher(pool, this.config.mpv3Enabled);
+    } else {
+      log.info("start() - Configuring single threaded dispatcher");
+      this.messageDispatcher = createSingleMessageDispatcher(config.mpv3Enabled);
```

Comment:
is the config same

Reply:
Yeah it sets up the V3 confgis based on the bool.

Reply:
I changed it to this.config so would not confuse.